src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -52,6 +60,8 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
             var scaffoldColumnAttribute = attributes.OfType<ScaffoldColumnAttribute>().FirstOrDefault();
             var uiHintAttribute = attributes.OfType<UIHintAttribute>().FirstOrDefault();
 
+            var localizer = _stringLocalizerFactory?.Create(context.Key.ContainerType ?? context.Key.ModelType);
```

Comment:
- Break the `context.Key.ContainerType ?? context.Key.ModelType` into its own variable to ease debuggability.
- The localizer is used specifically for the `DisplayAttribute`, move it down to where it's used so it's not created every time even if there is no display attribute :smile: .